core/core.js

```diff
@@ -213,24 +213,25 @@ SharedQueue Binary Layout
     throw new ErrorClass(res.err.message);
   }
 
-  async function jsonOpAsync(opName, args = {}, ...zeroCopy) {
+  async function jsonOpAsync(opName, args = null, ...zeroCopy) {
     setAsyncHandler(opsCache[opName], jsonOpAsyncHandler);
 
-    args.promiseId = nextPromiseId++;
-    const argsBuf = encodeJson(args);
-    dispatch(opName, argsBuf, ...zeroCopy);
+    const promiseId = nextPromiseId++;
+    const reqBuf = core.encode("\0".repeat(8) + JSON.stringify(args));
+    new DataView(reqBuf.buffer).setBigUint64(0, BigInt(promiseId));
```

Comment:
So the first 8 bytes represent the promiseId encoded as a little endian u64 ? Is that only the case for JSON ops or also for minimal ops?

Reply:
Yes, it's only for JSON ops. However, it looks like the two implementation are becoming more similar (#9457), so perhaps it would make sense to store the promiseId in the first 4 bytes as a big endian i32, like it is for minimal ops.

Reply:
That might allow them to share more code on the backend... I think it could be addressed in a follow up commit tho.